releasenotes/notes/add-bit-register-rust-979dd113aefb3563.yaml

```
@@ -0,0 +1,23 @@
+---
+features_circuits:
+  - |
+   Python classes :class:`.Bit` and :class:`.Register` have been ported to Rust
+   along with all their respective subclasses including:
+   
+   * :class:`.Qubit` and :class:`.QuantumRegister`.
+   * :class:`.Clbit` and :class:`.ClassicalRegister`.
+   * :class:`.AncillaQubit` and :class:`.AncillaRegister`.
+upgrade_circuits:
+  - |
+    :class:`.DAGCircuit` and :class:`.CircuitData` will now store :class:`.Qubit`,
+    :class:`.Clbit`, as well as its register counterparts, in Rust space.
+  - The :attr:`.QuantumCircuit._qubit_indices` and :attr:`.QuantumCircuit._clbit_indices`
+    attributes of the :class:`.QuantumCircuit` class have been moved to live in 
+    rust space and store rust native data.
+  - The :meth:`.QuantumCircuit._qbit_argument_conversion` and :meth:`.QuantumCircuit._cbit_argument_conversion`
+    have been moved to Rust to operate with native data.
+other:
+  - |
```

This should be an upgrade note because it's an api change that users will need to potentially update their usage for on upgrading.

```suggestion
  - |
```

---

Done in 0e669440a.